frontend/test/pytest/test_aws_braket_devices.py

```diff
@@ -0,0 +1,316 @@
+
+        expected = np.array(1.0)
+        observed = expval(0.0)
+        assert np.isclose(observed, expected)
+
+        expected = np.array(-1.0)
+        observed = expval(np.pi)
+        assert np.isclose(observed, expected)
+
+    def test_tensor(self):
+        """Test expval for Tensor observable on braket.aws.qubit."""
+
+        device = qml.device(
+            "braket.aws.qubit",
+            device_arn="arn:aws:braket:::device/quantum-simulator/amazon/sv1",
+            wires=2,
+        )
+
+        @qjit()
+        @qml.qnode(device)
+        def expval(x: float, y: float):
+            qml.RX(x, wires=0)
+            qml.RX(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliX(0) @ qml.PauliY(1))
+
+        expected = np.array(-0.35355339)
+        observed = expval(np.pi / 4, np.pi / 3)
+        assert np.isclose(observed, expected)
+
+        expected = np.array(-0.49999788)
+        observed = expval(np.pi / 2, np.pi / 3)
+        assert np.isclose(observed, expected)
+
+
+class TestBraketVar:
+    def test_rx(self):
+        """Test var with RX on braket.aws.qubit."""
+
+        device = qml.device(
+            "braket.aws.qubit",
+            device_arn="arn:aws:braket:::device/quantum-simulator/amazon/sv1",
+            wires=1,
+        )
+
+        @qjit()
+        @qml.qnode(device)
+        def var(x: float):
+            qml.RX(x, wires=0)
+            return qml.var(qml.PauliZ(0))
+
+        expected = np.array(0.0)
+        observed = var(0.0)
+        assert np.isclose(observed, expected)
+        observed = var(np.pi)
+        assert np.isclose(observed, expected)
+
+    def test_hadamard(self, backend):
+        """Test var with Hadamard on braket.aws.qubit."""
+
+        @qjit()
+        @qml.qnode(qml.device(backend, wires=1))
+        def var(x: float):
+            qml.Hadamard(wires=0)
+            return qml.var(qml.PauliZ(0))
+
+        expected = np.array(1.0)
+        observed = var(0.0)
+        assert np.isclose(observed, expected)
+        observed = var(np.pi)
+        assert np.isclose(observed, expected)
+
+    def test_tensor(self):
+        """Test variance for Tensor observable on braket.aws.qubit."""
+
+        device = qml.device(
+            "braket.aws.qubit",
+            device_arn="arn:aws:braket:::device/quantum-simulator/amazon/sv1",
+            wires=2,
+        )
+
+        @qjit()
+        @qml.qnode(device)
+        def circuit(x: float, y: float):
+            qml.RX(x, wires=0)
+            qml.RX(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return qml.var(qml.PauliX(0) @ qml.PauliY(1))
+
+        expected = np.array(0.875)
+        observed = circuit(np.pi / 4, np.pi / 3)
+        assert np.isclose(observed, expected)
+
+        expected = np.array(1.0)
+        observed = circuit(np.pi / 2, np.pi / 2)
+        assert np.isclose(observed, expected)
+
+
+if __name__ == "__main__":
+    pytest.main(["-x", __file__])
```

Comment:
Really nice tests @maliasadi! I have some questions:

- Do we need tests for Hamiltonians?
- Do we need tests for gradients?
- Do we need tests for probs?
- Do we need tests for when multiple returns are specified?
- What happens when we submit non-supported circuits? We should have tests to ensure the correct error message/behaviour is reflected for users

Reply:
> Do we need tests for Hamiltonians?

I've implemented hamiltonian in the backend! But seems not to be properly supported in the braket devices. I will investigate more.

> Do we need tests for gradients?

What do you think of adding this in the second version of this support as a followup PR in the future? I do have a WIP PR for this but wanted to finish it later after getting main pieces published as I expected this feature to be out of the scope of this ticket, but I can definitely work on it if this is something good to be out as part of the ticket.

> Do we need tests for probs?

added! 

> Do we need tests for when multiple returns are specified?

added!

> Do we need tests for when multiple returns are specified?

The user will see descriptive `RuntimeError` for not supported cases. Added a few tests.

Thanks @josh146! I'll add more tests while extending the support in the future (next versions of this support). Let me know if there's anything else missing.

Reply:
@josh146 yes, indeed! I wanted to address the gradient (both adjoint from the runtime and other methods) in a separate PR. With the current implementation, we do have support for param-shift ([added tests here](https://github.com/PennyLaneAI/catalyst/blob/746b903dd9a8a899124de42067a255f2c46b4fe8/frontend/test/pytest/test_aws_braket_devices.py#L576)) but finite-diff always returns 0 (haven't investigated the issue yet) and adjoint-jacobian (not fully supported - WIP).

Reply:
Thanks Ali! Regarding finite diff, my gut feeling/guess would be that the step is so low (is it `1e-7`?) that it could be rounded away at some point in the openqasm conversion.

Reply:
Smart! :sweat_smile: yeah, I totally forgot about the `h-`value and it's too low by default (`1e-7`). I'll add `fd` tests shortly! :)

Reply:
@josh146  finite-diff tests [c368910](https://github.com/PennyLaneAI/catalyst/pull/154/commits/c368910f2fd5507be883f3299ee2dfd6d3d2298d)